ipalib/x509.py

```diff
@@ -405,6 +405,18 @@ def match_hostname(self, hostname):
     def tbs_precertificate_bytes(self):
         return self._cert.tbs_precertificate_bytes
 
+    # added in python-cryptography 40.0
+    def verify_directly_issued_by(self, issuer):
+        """
+        This method verifies that certificate issuer name matches the
+        issuer subject name and that the certificate is signed by the
+        issuer's private key. No other validation is performed.
+        """
+        if 'verify_directly_issued_by' in self._cert.__abstract_methods__:
```

Comment:
What is `__abstract_methods__`? `__abstractmethods__`?

Reply:
`__abstractmethods__` tracks unimplemented abstract methods (that have `__isabstractmethod__` set to True). This condition is wrong.